widget/richtext.go

```diff
@@ -562,7 +562,20 @@ func (r *textRenderer) layoutRow(texts []fyne.CanvasObject, align fyne.TextAlign
 				realign = true
 			}
 			size = s
-		} else {
+		} else if c, ok := text.(*fyne.Container); ok {
+			wid := c.Objects[0]
+			if link, ok := wid.(*Hyperlink); ok {
+				s, base := fyne.CurrentApp().Driver().RenderedTextSize(link.Text, theme.TextSize(), link.TextStyle)
+				if tallestBaseline == 0 {
+					tallestBaseline = base
+				} else if base != tallestBaseline {
+					tallestBaseline = fyne.Max(tallestBaseline, base)
+					realign = true
+				}
+				size = s
+			}
+		}
```

Comment:
Hmm it is desired to support more kind of "texts", I think maybe it is better to reorganize this code to something like this:
```go
var (
  textStr string
  textSize float32
  textStyle TextStyle
  size fyne.Size
)
switch v := text.(type) {
   case *canvas.Text:
       textStr = v.Text
       textSize = v.TextSize
       textStyle = v.TextStyle
   case *fyne.Container:
       wid := c.Objects[0]
       if link, ok := wid.(*Hyperlink); ok {
           textStr = link.Text
           textSize = theme.TextSize()
           textStyle = link.TextStyle
       }
}
if textSize > 0 {
    	s, base := fyne.CurrentApp().Driver().RenderedTextSize(txt.Text, txt.TextSize, txt.TextStyle)
	if tallestBaseline == 0 {
		tallestBaseline = base
	} else if base != tallestBaseline {
	         tallestBaseline = fyne.Max(tallestBaseline, base)
		realign = true
	}
        s = size
} else {
        size = text.MinSize()
}
```
But I am not totally sure if this is clearer or not, what do you think?

Reply:
In the future it may be desirable to find a way to make a more generalised alignable API for a RichText element - but I think it is too early to know how, so I just added these two known types for now.

Reply:
Got it, my above suggestion may improve this too :), but agree that could be a future improvement, although I think it is relatively easy to add it now 🧐